cerbero/build/source.py

```diff
@@ -408,14 +409,103 @@ class Git (GitCache):
     Source handler for git repositories
     '''
 
+    # Maps recipes with common sources share to the same fetch_done condition.
+    _fetch_done_cond_map = {}
+
     def __init__(self):
         GitCache.__init__(self)
         if self.commit is None:
             # Used by recipes in recipes/toolchain/
             self.commit = 'origin/sdk-%s' % self.version
+
+        self._fetching = False
         # For forced commits in the config
         self.commit = self.config.recipe_commit(self.name) or self.commit
 
+    def __getattribute__(self, attr_name):
+        attr = super().__getattribute__(attr_name)
+        if attr_name == "prepare":
+            self.__git_prepare = attr
+            raise AttributeError
+        return attr
+
+    def __getattr__(self, attr):
+        if attr == "prepare":
+            def _prepare():
+                # Set the fetch condition object after the default prepare
+                # method, as some recipes (properly) set/change remotes or
+                # repo_dir attributes from the prepare method, and the
+                # _update_fetch_done_cond method needs this info.
+                self.__git_prepare()
+                self._update_fetch_done_cond()
```

Comment:
I think this can be called in the fetch instead of doing this complicated thing just to make sure we do it after the prepare. Please give it a thought to see if we can make it simpler